### ISSUE TITLE
Release 0.1: Roadmap

### ISSUE THREAD

**Issue by ejgallego** [ejgallego/coq-universe#37]
Hi folks ,

we had some discussion today with @gbdrt and Laetitia Teodorescu (@laetitia-teo) about getting Coq Universe back to a functional state for us.

I think that could be a good time to actually announce a first release of Coq Universe, let's say 0.1.

Here is what I understood we need to do before that can happen:
- better opam handling: I think we need:
  + make Coq Universe run over a local opam repos
  + provide an opam lock file so we are fully reproducible
  + maybe pin our fork of Dune for that opam repos? That would solve a few problems, but introduce latency for others
- better submodules handling: Here we have a few items:
  + submodules are painful to handle in different forks or users, in particular when people want to have a private fork, things are not good. What I propose is that we create a coq-universe organization and require that we host a fork of each project in our org. That's not perfect, but IMHO better and provides a more reproducible setup. Thoughts @Alizter ?
  + submodules helpers, see #9 , we can host it in the coq-universe org actually!
- update submodules to newer versions: I'm working on it
- other items, such as silent builds.

What do you think folks?

**Comment by Alizter** [comment]
With a new version of Dune, we should see if we can get Coq to build using the package management infrastructure. That would let us build various Coq projects even if they don't use Dune. That would also let us avoid submodules.

However I suspect that there are some issues getting this working, especially with how Coq installs theories which means it may not be viable.

**Comment by ejgallego** [comment]
@Alizter , that would be great indeed!

I suggest we track progress on that on its own issue, as it seems to me that is more of a long term plan.

One question I have about the current package support, is if it can replace the current pattern in coq-universe to build a concrete file as fast as possible, that is to say, for example `dune build math-comp/math-comp/field/alg_count.vo` ?

**Comment by ejgallego** [comment]
@Alizter I assume the rest of the roadmap is making sense to you, so I'll go ahead with the part about creating an org and storing the repos there by default. Let me know if you would like to do differently.

**Comment by Alizter** [comment]
>One question I have about the current package support, is if it can replace the current pattern in coq-universe to build a concrete file as fast as possible, that is to say, for example dune build math-comp/math-comp/field/alg_count.vo ?

No, because each package build would be happening in a directory target. In package management, there is only a "single rule" building a package, therefore you won't be able to get fine-grained targets without building the entirety of mathcomp.

In order to get that, you would need to vendor a build, and I suppose this is what splits coq-universe from a regular opam CI build.

One thing package management could help with is inserting non-dune builds in the dependency cones. This would help greatly in maintenance, especially for projects that don't use a dune build. We would probably have to patch the opam files still, as the install location in user-contrib is not appropriate. Something like its own folder and an addition to COQPATH should be enough.

**Comment by ejgallego** [comment]
I see, thanks. A core use case of coq-universe tho is to be able to access a particular file as fast as possible, so indeed, for now it seems to me that the only option is the composed dune build.

The whole point of Coq Universe for me is to remain incremental.

I think it is easy to migrate projects to Dune, once we reach 1.0 version of the language tho.
